Objects/complexobject.c

```diff
@@ -1030,12 +1044,7 @@ complex_new_impl(PyTypeObject *type, PyObject *r, PyObject *i)
         }
         if (tmp == NULL)
             return NULL;
-        if (!PyFloat_Check(tmp)) {
-            PyErr_SetString(PyExc_TypeError,
-                            "float(r) didn't return a float");
-            Py_DECREF(tmp);
-            return NULL;
-        }
+        assert(PyFloat_Check(tmp));
```

Comment:
It could be included in bpo-26983, but it was not. This PR is good opportunity to make this change since it is related to `complex.__new__`.

Do you suggest to exclude this change from the PR and make a separate PR.